corporate/lib/stripe.py

```diff
@@ -4989,7 +4989,12 @@ def invoice_plans_as_needed(event_time: Optional[datetime] = None) -> None:
                     plan.save(update_fields=["invoice_overdue_email_sent"])
                 continue
 
-        billing_session.invoice_plan(plan, plan.next_invoice_date)
+        while (
+            plan.next_invoice_date is not None  # type: ignore[redundant-expr] # plan.next_invoice_date can be None after calling invoice_plan.
```

Comment:
I don't think `plan.refresh_from_db` can actually result in `plan` being `None`. So I think something is confused here.

Reply:
`plan.next_invoice_date` can be none if customer plan has ended.

```python
def next_invoice_date(plan: CustomerPlan) -> Optional[datetime]:
    if plan.status == CustomerPlan.ENDED:
        return None
    assert plan.next_invoice_date is not None  # for mypy
    months_per_period = 1
    periods = 1
    dt = plan.billing_cycle_anchor
    while dt <= plan.next_invoice_date:
        dt = add_months(plan.billing_cycle_anchor, months_per_period * periods)
        periods += 1
    return dt
```

Reply:
Oh, yeah, I am the thing that's confused -- was thinking it was `plan is not None` for some reason.